src/xclim/indices/_threshold.py

```diff
@@ -2014,6 +2075,61 @@ def hot_spell_max_magnitude(
     return to_agg_units(out, tasmax, op="integral")
 
 
+@declare_units(tasmax="[temperature]", tasmin="[temperature]", tas="[temperature]", thresh="[temperature]")
+def heating_degree_days_approximation(
+    tasmax: xarray.DataArray,
+    tasmin: xarray.DataArray,
+    tas: xarray.DataArray,
+    thresh: Quantified = "17.0 degC",
+    freq: str = "YS",
+) -> xarray.DataArray:
+    """
+    Heating degree days approximation.
+
+    A more robust approximation of heating degree days as a function of the daily cycle of temperature.
+
+    Parameters
+    ----------
+    tasmax : xarray.DataArray
+        Maximum daily temperature.
+    tasmin : xarray.DataArray
+        Minimum daily temperature.
+    tas : xarray.DataArray
+        Mean daily temperature.
+    thresh : Quantified
+        Threshold temperature on which to base evaluation.
+    freq : str
+        Resampling frequency.
+
+    Returns
+    -------
+    xarray.DataArray
+        Approximation of heating degree days.
+
+    References
+    ----------
+    :cite:cts:`spinoni_2018`
+    """
+    # Where tasmax <= thresh; HDD = thresh - tas
+    # Where tas <= thresh < tasmax; HDD = (thresh - tasmin)/2 - (tasmax - thresh)/4
+    # Where tasmin < thresh < tas; HDD = (thresh - tasmin)/4
+    # Where tasmin >= thresh; HDD = 0
+    thresh = convert_units_to(thresh, tasmax)
+
+    hdd = xarray.where(
+        tasmax <= thresh,
+        thresh - tas,
+        xarray.where(
+            tas <= thresh,
+            (thresh - tasmin) / 2 - (tasmax - thresh) / 4,
+            xarray.where(tasmin <= thresh, (thresh - tasmin) / 4, 0),
+        ),
+    )
+    hdd = hdd.resample(time=freq).sum(dim="time")
+    hdd.attrs.update(units=tas.units)
```

Comment:
```suggestion
    hdd = to_agg_units(hdd, tas, 'integral')
```
Similar, but here you were missing the temporal part!